### PR TITLE
Updates to Operator beta docs

### PR DESCRIPTION
**Pull request by JakeSCahill**
Part of #1624 

- Updated the Redpanda Operator image to rc7
- Added a deprecated features reference page
- Added troubleshooting steps for Redpanda Operator
- Added the required `chartRef` field in the CR (won't be required for GA when this is merged https://github.com/redpanda-data/redpanda/pull/11877)
- Replaced hardcoded namespaces with a placeholder to allow users to choose the namespace in which to deploy Redpanda

Previews:

- [Troubleshooting](https://deploy-preview-1753--redpanda-documentation.netlify.app/docs/beta/manage/kubernetes/troubleshooting/troubleshoot/#helmrelease-is-not-ready)
- [Deprecated Features](https://deploy-preview-1753--redpanda-documentation.netlify.app/docs/beta/reference/deprecated/)
- [Deployment](https://deploy-preview-1753--redpanda-documentation.netlify.app/docs/beta/deploy/deployment-option/self-hosted/kubernetes/kubernetes-deploy/#deploy-redpanda)

### CODE REVIEW

**Review by joejulian — APPROVED** [review]
Approving with one suggestion

File: versioned_docs/version-23.2/reference/deprecated/index.mdx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+<head>
+    <meta name="title" content="Deprecated Features"/>
+    <meta name="description" content="This guide helps you to identify deprecated features in Redpanda 23.2 and plan necessary upgrades or modifications."/>
+</head>
+
+This guide helps you to identify deprecated features in Redpanda 23.2 and plan necessary upgrades or modifications.
+
+## Deprecated: Cluster and Console custom resources
+
+As of Redpanda version 23.2, the Cluster and Console custom resources are deprecated.
```
Comment (alejandroEsc) [inline versioned_docs/version-23.2/reference/deprecated/index.mdx:15]: nice!

**Review by alejandroEsc — APPROVED** [review]
lgtm!

File: versioned_docs/version-23.2/reference/redpanda-operator/kubernetes-external-connect.mdx [hunk 2]
```diff
@@ -148,13 +148,13 @@ To use Helm to install cert-manager:
 
     ```bash
     kubectl apply -k https://github.com/redpanda-data/redpanda/src/go/k8s/config/crd?ref=$VERSION && \
-    helm repo add redpanda https://charts.vectorized.io/ && \
+    helm repo add vectorized https://charts.vectorized.io/ && \
```
Comment (Deflaimun) [inline versioned_docs/version-23.2/reference/redpanda-operator/kubernetes-external-connect.mdx:151]: Is that change correct?

**Review by shorgi — APPROVED** [review]

File: versioned_docs/version-23.2/reference/redpanda-operator/kubernetes-external-connect.mdx (hunk 2 above)
Comment (JakeSCahill) [reply versioned_docs/version-23.2/reference/redpanda-operator/kubernetes-external-connect.mdx:151]: the old Operator won't be moved from `vectorized` so this change avoids naming conflicts with the `redpanda` chart repo

**Pull request merged**